test/NuGet.Core.Tests/NuGet.Packaging.Test/SigningTests/CertificateUtilityTests.cs

```diff
@@ -223,10 +233,12 @@ public void IsValidForPurposeFast_WithAnyExtendedKeyUsage_ReturnsFalseForCodeSig
             using (var certificate = SigningTestUtility.GenerateCertificate("test",
                 generator =>
                 {
-                    generator.AddExtension(
-                        X509Extensions.ExtendedKeyUsage.Id,
-                        critical: false,
-                        extensionValue: new ExtendedKeyUsage(KeyPurposeID.IdKPEmailProtection, KeyPurposeID.AnyExtendedKeyUsage));
+                    var usages = new OidCollection { TestCertificateGenerator.IdKPEmailProtection, TestCertificateGenerator.AnyExtendedKeyUsage };
+
+                    generator.Extensions.Add(
+                        new X509EnhancedKeyUsageExtension(
+                              usages,
+                              critical: true));
```

Comment:
`critical` should be `false`.  From https://tools.ietf.org/html/rfc5280#section-4.2.1.12:
>If a CA includes extended key usages to satisfy such applications,
   but does not wish to restrict usages of the key, the CA can include
   the special KeyPurposeId anyExtendedKeyUsage in addition to the
   particular key purposes required by the applications.  Conforming CAs
   SHOULD NOT mark this extension as critical if the anyExtendedKeyUsage
   KeyPurposeId is present.